makemkv/types.py

```diff
@@ -22,31 +23,47 @@ class Drive(TypedDict, total=False):
 
 
 class Disc(TypedDict, total=False):
+    comment: str
     information: str
+    metadata_langcode: str  # two-letter ISO 639-1 code if it exists, eg. "en"
+    metadata_language: str  # eg. "English"
     name: str
     type: Literal["DVD", "BD", "HDDVD", "MKV"]
+    volume_name: str
 
 
 class Stream(TypedDict, total=False):
     aspect_ratio: str  # eg. "16:9"
     bitrate: str  # eg. "384 Kb/s"
-    codec: str  # eg. "V_MPEG2", "A_AC3", "S_VOBSUB"
+    codec_id: str  # eg. "V_MPEG2", "A_AC3", "S_VOBSUB"
+    codec_long: str  # eg. "Mpeg4 AVC High@L4.1"
+    codec_short: str  # eg. "Mpeg4"
     dimensions: str  # eg. "720x576",
     downmix: str  # eg. "Surround 5.1"
-    framerate: int  # eg. 25
+    framerate: int | Decimal  # eg. 25 or Decimal("23.976")
```

Comment:
I requires python 3.10

Reply:
i changed it to use Union instead

Reply:
also Decimal is not compatable with json
![image](https://user-images.githubusercontent.com/61570822/172772132-38f70575-1839-42e2-b8f3-2e6d285b5fd0.png)

Reply:
Oh, you are absolutely right. I forgot about Python 3.9. This also means that the validation function needs a special case for `Union` (which wouldn't be required on Python 3.10).

Furthermore, I wasn't aware that 23.976 isn't the precise framerate. So unless we want to use something like `fractions.Fraction(24000, 1001)` (which would be more confusing than helpful) it may be better to go back to `float`. The potential floating point errors seem to be negligible.